Demo/Demo/Views/CustomButton.swift

```diff
@@ -18,4 +30,29 @@ class CustomButton: UIButton {
     required init?(coder: NSCoder) {
         super.init(coder: coder)
     }
+
+    func startAnimating() {
+        self.setTitle("", for: .normal)
+
+        showSpinning()
+    }
+
+    func stopAnimating() {
+        self.setTitle(titleHolder, for: .normal)
+        activityIndicator.stopAnimating()
+    }
```

Comment:
one concern I have is that we keep adding subview + setting constraints for the `activityIndicator` everytime the button is tapped.
What do you think of doing `addSubview()` and `setupConstraints()` in `init()`, then toggle either `alpha` or `isHidden` of `activityIndicator` in these functions? :smile:

Reply:
Yep, I think it makes sense to move add subview and setup constraints in the init. For the alpha, I think it makes sense to use the `start/stopAnimating` functions built into activity indicator as that will always give us the expected behavior, but that can be simplified once we move the other portion to the init.